main.py

```diff
@@ -0,0 +1,68 @@
+import uvicorn
+from typing import List
+from opencdms.models.climsoft.v4_1_1_core import Station
+from fastapi import FastAPI, Depends
+from sqlalchemy.orm import sessionmaker
+from sqlalchemy import create_engine
+from sqlalchemy.orm.session import Session
+from pydantic import BaseModel
+
+app = FastAPI()
+
+# Set database url to mariadb_climsoft_db_v4 database url
+database_url = ""
+engine = create_engine(database_url)
+
+
+SessionLocal = sessionmaker(engine)
+
+
+
+class StationSchema(BaseModel):
```

Comment:
@isedwards  FastAPI doesn't support this out of the box. However, there is an experimental library for this feature https://github.com/tiangolo/pydantic-sqlalchemy. I don't see an issue with having a separate schema model. Yes it might be similar to the db model but I'd argue that it's easier to maintain, more flexible and more powerful. I'd rather do this than have the introspection (I think😅).

Reply:
Okay, I'm happy with this. Looking again it's actually not much more verbose that maintaining a whitelist of which fields to expose.